cloudant-client/src/test/java/com/cloudant/tests/HttpTest.java

```diff
@@ -345,6 +345,91 @@ public void cookieRenewal() throws Exception {
     }
 
     /**
+     * This test check that the cookie is renewed if the server presents a Set-Cookie header
+     * after the cookie authentication.
+     *
```

Comment:
It might be worth documenting the flow expected in this test. As it stands I don't think it is valid. If the (couch) server sends a new `Set-Cookie` then it is effectively providing us with a new session, under which circumstances we should have no need to contact the IAM server. I don't think that an IAMSession will ever be renewed in that way, but if it was we should just use the new cookie, i.e. we should assert 0 `_iam_session` requests.

As stated in the PR description I think more tests are needed. Equivalent mock tests for doing:
* a successful initial session
* a successful renewal on a 401
* failure to get a token initially
* failure to get a token on renewal